trackfilter.cc

```diff
@@ -57,90 +72,54 @@ int TrackFilter::trackfilter_opt_count()
   return res;
 }
 
-int TrackFilter::trackfilter_parse_time_opt(const char* arg)
+qint64 TrackFilter::trackfilter_parse_time_opt(const char* arg)
 {
-  int sign = 1;
-  const char* cin = arg;
-  char c;
-
-  time_t t0 = 0;
-  time_t t1 = 0;
-
-  while ((c = *cin++)) {
-    time_t seconds;
-
-    if (c >= '0' && c <= '9') {
-      t1 = (t1 * 10) + (c - '0');
-      continue;
+  qint64 result;
+
+  QRegularExpression re("^([+-]?\\d+)([dhms])$", QRegularExpression::CaseInsensitiveOption);
+  assert(re.isValid());
+  QRegularExpressionMatch match = re.match(arg);
+  if (match.hasMatch()) {
+    bool ok;
```

Comment:
Do our warning tools know that ok won't be used uninitialized? Interesting.

Reply:
I haven't found any complaints, but there is always another checker to check.